var/spack/repos/builtin/packages/heffte/package.py

```diff
@@ -97,59 +97,10 @@ def cmake_args(self):
 
         return args
 
-    @run_after('install')
-    def cache_test_sources(self):
-        """Copy the example source files after the package is installed to an
-        install test subdirectory for use during `spack test run`."""
-        self.cache_extra_test_sources([self.examples_src_dir])
-
     def test(self):
-        cmake_file = join_path(self.install_test_root, 'CMakeLists.txt')
-        test_bld_dir = join_path(self.install_test_root, '_build')
-        exe_files = []
-        mpi_procs = []
-
-        with open(cmake_file, 'w') as cmkf:
-            cmkf.write('cmake_minimum_required(VERSION 3.13)\n\n')
-            cmkf.write('project(heffte_example LANGUAGES CXX)\n\n')
-            cmkf.write('find_package(Heffte REQUIRED)\n\n')
-            if '+fftw' in self.spec:
-                src_file = '{0}/heffte_example_fftw.cpp'.format(
-                    self.examples_src_dir)
-                cmkf.write(
-                    'add_executable(hf_fftw {0})\n'.format(src_file))
-                cmkf.write(
-                    'target_link_libraries(hf_fftw Heffte::Heffte)\n')
-                exe_files.append('./hf_fftw')
-                mpi_procs.append(4)
-            if '+cuda' in self.spec:
-                src_file = '{0}/heffte_example_cuda.cpp'.format(
-                    self.examples_src_dir)
-                cmkf.write(
-                    'add_executable(hf_cuda {0})\n'.format(src_file))
-                cmkf.write(
-                    'target_link_libraries(hf_cuda Heffte::Heffte)\n')
-                exe_files.append('./hf_cuda')
-                mpi_procs.append(4)
-            if '+fortran' in self.spec:
-                src_file = '{0}/heffte_example_fftw.f90'.format(
-                    self.examples_src_dir)
-                cmkf.write(
-                    'add_executable(hf_fort {0})\n'.format(src_file))
-                cmkf.write(
-                    'target_link_libraries(hf_fort Heffte::Fortran)\n')
-                exe_files.append('./hf_fort')
-                mpi_procs.append(2)
-
-        with working_dir(test_bld_dir, create=True):
-            cmake('..')
+        # using the tests installed in <prefix>/share/heffte/testing
+        cmake_dir = join_path(self.prefix, 'share', 'heffte', 'testing')
+        with working_dir(self.install_test_root, create=True):
```

Comment:
Building within any install root directory has permissions issues with facility deployments and shared Spack instances.

Is it at all possible to do this from`self.test_suite.current_test_cach_dir`?  You would need to create the directory -- or an alternate subdirectory of it if you are no longer using `cache_test_sources`.  For example:

```suggestion
        with working_dir(self.test_suite.current_test_cache_dir, create=True):
```

If you have issues with the directory we may want to add a new property that gives ready access to a suitable alternative.

Reply:
I changed the folder to a sub-folder of `self.test_suite.current_test_cache_dir`

All the test needs is a clean folder to run CMake.